langlib/lang.xml/src/main/java/org/ballerinalang/langlib/xml/Map.java

```diff
@@ -60,7 +61,13 @@ public static BXml map(BXml x, BFunctionPointer<Object, Object> func) {
         AtomicInteger index = new AtomicInteger(-1);
         AsyncUtils.invokeFunctionPointerAsyncIteratively(func, null, METADATA, x.size(),
                 () -> new Object[]{x.getItem(index.incrementAndGet()), true},
-                result -> elements.add((BXml) result), () -> ValueCreator.createXmlSequence(elements),
+                result -> {
+                    if (result instanceof XmlSequence xmlSequence) {
```

Comment:
I have a concern about implementing this logic here. I think this logic should be within the `createXmlSequence` function. If not, every time we create an XML list, we will need to check whether it is a sequence and then add its children.

Please check the usage of `createXmlSequence` as well. There might be other places where we haven't applied this logic.